crates/katana/rpc/rpc/src/torii.rs

```diff
@@ -0,0 +1,173 @@
+use std::sync::Arc;
+
+use futures::StreamExt;
+use jsonrpsee::core::{async_trait, RpcResult};
+use katana_core::sequencer::KatanaSequencer;
+use katana_core::service::block_producer::BlockProducerMode;
+use katana_primitives::block::BlockHashOrNumber;
+use katana_provider::traits::transaction::TransactionProvider;
+use katana_rpc_api::torii::ToriiApiServer;
+use katana_rpc_types::error::torii::ToriiApiError;
+use katana_rpc_types::receipt::{MaybePendingTxReceipt, PendingTxReceipt};
+use katana_rpc_types::transaction::{TransactionsPage, TransactionsPageCursor};
+use katana_rpc_types_builder::ReceiptBuilder;
+use katana_tasks::TokioTaskSpawner;
+
+const MAX_PAGE_SIZE: usize = 100;
```

Comment:
having a global limit is good, but shouldn't we also include a chunk size in the request param itself ? I'm aware that this endpoint will prob be used by `torii` mostly for now but imo adding a chunk size would encourage doign smaller requests in general.

Reply:
i think we should add this as a follow up. we can also introduce a filter for txns relevant to the world